### ISSUE TITLE
tpl: in function should check for nil params

### ISSUE THREAD

**Issue by moorereason** [spf13/hugo#3225]
Panic with this test:

```
{[]string{"a", "b", "c"}, nil, false},
```

Other types will panic, too.

See https://discuss.gohugo.io/t/5903

**Comment by ajiyakin** [comment]
Hi @moorereason, I'm a new comer.
Does this issue good for new comer like me? If so, I will try to solve this issue.

Basically the `in` function should be returned an `error` and handled by it's caller right?

**Comment by bep** [comment]
>Basically the in function should be returned an error and handled by it's caller right?

No, I think the expected behaviour would be to return false in the test case below:

```
{[]string{"a", "b", "c"}, nil, false},
```

As `nil` is not `in` that list.

That said, it would be great if you could fix it.

**Comment by ajiyakin** [comment]
Oh I see... This should be easy.
I see why this test fail, because for the other type already has a checker whether it has same type or not (`lv` with `vv`), in case of string, it doesn't.

I already fix it. I'll do a pull request :+1:

**Comment by moorereason** [comment]
Fixed in #3177.
